### PR TITLE
Feature/update hardhat schema

### PR DESCRIPTION
**Pull request by compojoom**

### CODE REVIEW

File: deploy/development/docker-compose.yml [hunk 1]
```diff
@@ -14,11 +14,17 @@ services:
...
+      ethereum: "pow:${POW_ETHURL} pos:${POS_ETHURL}"
       GRAPH_LOG: info
+      # can be "index-node", "query-node" or "combined-node"
+      node_role: "${GRAPH_NODE_ROLE}"
+      node_id: "${GRAPH_NODE_ID}"
     labels:
-      caddy: graph.staging.powdump.com
+      caddy: localhost
       caddy.reverse_proxy: "{{upstreams 8000}}"
+      caddy.reverse_proxy.header_down_1: +Access-Control-Allow-Methods "GET, POST, PATCH, OPTIONS"
```
Comment (ezdac) [inline deploy/development/docker-compose.yml:25]: Why did you add the response headers? Is this also required in production or only relevant for local development?

File: deploy/development/docker-compose.yml [hunk 2]
```diff
@@ -28,32 +34,36 @@ services:
...
+      dockerfile: Dockerfile
+      context: ../..
+    working_dir: /usr/app/checkout/packages/contracts
     entrypoint:
       - npx
       - hardhat
     volumes:
-      - ./data/deployments:/usr/app/packages/contracts/deployments
+      - ./data/contracts:/usr/app/checkout/packages/contracts/artifacts
+      - ./data/deployments:/usr/app/checkout/packages/contracts/deployments
```
Comment (ezdac) [inline deploy/development/docker-compose.yml:65]: As we saw before, this being a child directory of the volume mapping below will cause the checked out `contracts/deployments` dir to be created upon `hardhat deploy` if it didn't exist. Maybe it will even be overwritten if it exists. I think it would be better to configure the paths more considerately with the help of [hardhot-deploy-plugin path options](https://github.com/wighawag/hardhat-deploy/tree/master#3-extra-hardhatconfig-paths-options). In any case this is not critical currently since we don't rely on reading the deployment info from this location.

**Review by ezdac — APPROVED** [review]
Interested in learning about the Access-Control headers, but apart from that LGTM

File: deploy/development/docker-compose.yml (hunk 1 above)
Comment (compojoom) [reply deploy/development/docker-compose.yml:25]: The header is there because without it I was getting a CORS errors. I don't think it is required for production as there we access everything over https.
